storage/storage_image.go

```diff
@@ -330,6 +366,16 @@ func (s *storageImageDestination) Commit() error {
 			lastLayer = layerList[len(layerList)-1]
 		}
 	}
+	if s.ID == "" {
```

Comment:
Why can't we always use the config blob digest? It should be present for every image afaict and will allow us to group tags by id

Reply:
schema1 images, like `kubernetes/pause`, don't have config blobs.

Reply:
Mmm how is docker calculating the correct content addressable image ID for schema 1 images? We could copy that

Reply:
Docker converts the schema1 manifest into a config, and uses the digest of that as an ID.

I think the `c/i/image` conversion is very similar, but it’s not tested to generate exactly the same IDs, and I don’t even know whether there is any expectation of this conversion to be stable and frozen forever; I’d be a bit surprised if it were.